core/src/test/java/com/devonfw/tools/solicitor/componentinfo/curating/ComponentInfoCuratorTest.java

```diff
@@ -0,0 +1,120 @@
+package com.devonfw.tools.solicitor.componentinfo.curating;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.devonfw.tools.solicitor.componentinfo.ComponentContentProvider;
+import com.devonfw.tools.solicitor.componentinfo.ComponentInfo;
+import com.devonfw.tools.solicitor.componentinfo.ComponentInfoAdapterException;
+import com.devonfw.tools.solicitor.componentinfo.curation.ComponentInfoAdapterNonExistingCurationDataSelectorException;
+import com.devonfw.tools.solicitor.componentinfo.curation.ComponentInfoCurator;
+import com.devonfw.tools.solicitor.componentinfo.curation.ComponentInfoCuratorImpl;
+import com.devonfw.tools.solicitor.componentinfo.curation.CurationProvider;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+/**
+ * Unit tests for the {@link ComponentInfoCurator} class.
+ */
+public class ComponentInfoCuratorTest {
+
+  private ComponentInfoCurator componentInfoCurator;
+
+  @BeforeEach
+  public void setup()
+      throws ComponentInfoAdapterNonExistingCurationDataSelectorException, ComponentInfoAdapterException {
+
+    // Mocking required dependencies
+    CurationProvider curationProvider = mock(CurationProvider.class);
+    when(curationProvider.findCurations("pkg/maven/somenamespace/somecomponent/2.3.4", "none")).thenReturn(null);
```

Comment:
The handling of the special value "none" is currently neither specified nor done within "CurationProvider.findCurations()". This might be an alternative approach (see my corresponding comment), but the handling of "none" is currently done in the methods calling CurationProvider.findCurations(). If you now mock this behaviour in CurationProvider.findCurations(). this does not reflect the real behaviour of that class and the test might produce missleading results. (Actually the test would be successful even if there was not change in ComponentInfoCuratorImpl done.)